api/v0/finder/client/http/client.go

```diff
@@ -8,21 +8,25 @@ import (
 	"io"
 	"net/http"
 
+	"github.com/ipni/go-indexer-core"
+	"github.com/ipni/go-indexer-core/store/dhash"
 	"github.com/ipni/storetheindex/api/v0/finder/model"
 	"github.com/ipni/storetheindex/api/v0/httpclient"
 	"github.com/libp2p/go-libp2p/core/peer"
+	b58 "github.com/mr-tron/base58/base58"
 	"github.com/multiformats/go-multihash"
 )
 
 const (
-	finderPath    = "/multihash"
+	finderPath    = "/private/multihash"
```

Comment:
Client has been changed to serve private requests only. We can have multiple flavours of the client to server both legacy and new lookups.

Reply:
This is a bad idea; we should infer this from the multihash code of the multihash.

Reply:
What do you mean? Client always takes the original multihash (not the second hash).

Reply:
Server will return different responses depending on whether it's a regular / double hashed lookup. So we can't possibly have a single endpoint serving both. Can you elaborate please?

Reply:
There is a [dedivated multicodec](https://github.com/multiformats/go-multicodec/blob/master/code_table.go#L124) for double hashed multihashes. Such multihashes are used for private lookup. We should infer the type of query by parting the multihash and based on the code respond appropriately. 

> So we can't possibly have a single endpoint serving both

Why?

Reply:
They are structurally different

Reply:
Let's circle back on this at colo.

Reply:
Discussed in colo: 

We agreeded that 
* `/mh/<multihash>` would support both and change response based on the code in the given multihash.
* if the code is double-hashed, then the returned JSON object would include `EncryptedMultihashResults` which maps to an array of base64 encoded bytes instead of the regular `MultihashResults`.
* if the code is not double-hashed then no change.

This also facilitates response to POST requests that may contain a mix of double hashed and regular multihashes.

Reply:
So, it seems that the question here is which implementation do we want:
1.  The client queries the same endpoint on the server, sending either the plain or the hashed MH to the server, and the server responds with the appropriate response based on what type of MH was given.
2. The client queries a separate endpoint on the server, depending on whether it is sending a plain or a hashed MH. Those different endpoints respond with the respective response types.

Reply:
Will there ever be a case where there is a mixed set of plain and hashed MHs?

Having separate endpoints for plain and hashed MH queries has the advantage of being able to disable one or the other to support only one. Having separate endpoints also allows routing to different servers, which may be useful if double-hashing is supported on one and not another.